advanced/wallets/react-wallet-v2/src/views/SessionProposalModal.tsx

```diff
@@ -246,39 +229,54 @@ export default function SessionProposalModal() {
 
       try {
         await web3wallet.approveSession({
-          id,
-          relayProtocol: relays[0].protocol,
+          id: proposal.id,
           namespaces
         })
       } catch (e) {
+        setIsLoadingApprove(false)
         styledToast((e as Error).message, 'error')
         return
       }
     }
+    setIsLoadingApprove(false)
     ModalStore.close()
-  }
+  }, [proposal, supportedNamespaces])
 
   // Hanlde reject action
-  async function onReject() {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const onReject = useCallback(async () => {
     if (proposal) {
       try {
+        setIsLoadingReject(true)
+        await new Promise(resolve => setTimeout(resolve, 1000))
         await web3wallet.rejectSession({
-          id,
+          id: proposal.id,
           reason: getSdkError('USER_REJECTED_METHODS')
         })
       } catch (e) {
+        setIsLoadingReject(false)
         styledToast((e as Error).message, 'error')
         return
       }
     }
+    setIsLoadingReject(false)
     ModalStore.close()
-  }
+  }, [proposal])
 
-  return (
+  return !proposal ? (
```

Comment:
One thing I noticed here is that reusing a spent pairing URI will keep this in infinite load but I remember us discussing yesterday this really being more of a protocol-level problem with awaiting the peer response right?

Reply:
it would actually keep the loading for ~5 mins until the pairing expires and show an error message. It's not ideal but for used pairings, we can't do much. Once improvement that we could do, is track used pairings on the instance and throw immediately if we already used it. If pairing was used on different instance/wallet, we're stuck with current approach